File: backend/src/main/java/csc13001/plantpos/domain/models/DiscountProgram.java
package csc13001.plantpos.domain.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonInclude(JsonInclude.Include.NON_NULL)
@Entity
@Table(name = "discount_programs")
public class DiscountProgram {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "discount_id")
    private Long discountId;

    @NotBlank(message = "Name is mandatory")
    @Column(name = "name")
    private String name;

    @NotNull(message = "Start date is mandatory")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Column(name = "start_date")
    private Date startDate;

    @NotNull(message = "End date is mandatory")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @Column(name = "end_date")
    private Date endDate;

    @NotBlank(message = "Applicable customer type is mandatory")
    @Column(name = "applicable_customer_type")
    private String applicableCustomerType;
}

File: csc13001-plant-pos.Core/Models/DiscountProgram.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csc13001_plant_pos.Core.Models;

[Table("discount_programs")]
public class DiscountProgram
{
    [Key]
    [Column("discount_id")]
    public int DiscoundId
    {
        get; set;
    }

    [Column("name")]
    public string Name
    {
        get; set;
    }

    [Column("start_date")]
    public DateTime StartDate
    {
        get; set;
    }

    [Column("end_date")]
    public DateTime EndDate
    {
        get; set;
    }

    [Column("applicable_customer_type")]
    public string ApplicableCustomerType
    {
        get; set;
    }
}
